File: starters/tken-python-openai-compatible-cookbook/examples/route_selector.py
import json
import os
import re

LOW_COST_MODEL = os.getenv("LOW_COST_MODEL", "free-model")
PREMIUM_MODEL = os.getenv("PREMIUM_MODEL", "premium-gpt")


def choose_model(task: str) -> str:
    hard = re.search(r"debug|security|multi-file|architecture|failing test", task, re.I)
    return PREMIUM_MODEL if hard else LOW_COST_MODEL


if __name__ == "__main__":
    task = "debug failing tests"
    print(json.dumps({
        "task": task,
        "model": choose_model(task),
        "base_url": os.getenv("TKEN_BASE_URL", "https://www.tken.shop/v1"),
    }, indent=2))

File: starters/tken-node-openai-compatible-cookbook/examples/route-selector.mjs
const lowCostModel = process.env.LOW_COST_MODEL || "free-model";
const premiumModel = process.env.PREMIUM_MODEL || "premium-gpt";

export function chooseModel(task) {
  return /debug|security|multi-file|architecture|failing test/i.test(task) ? premiumModel : lowCostModel;
}

const task = process.argv.slice(2).join(" ") || "debug failing tests";
console.log(JSON.stringify({ ok: true, task, model: chooseModel(task), baseUrl: process.env.TKEN_BASE_URL || "https://www.tken.shop/v1" }, null, 2));
